BrickKit/app/build.gradle

```diff
@@ -29,23 +27,23 @@ android {
 }
 
 dependencies {
-    androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
-        exclude group: 'com.android.support', module: 'support-annotations'
-    })
-
-    compile "com.android.support:support-v4:${androidSupportVersion}"
-    compile "com.android.support:support-v13:${androidSupportVersion}"
-    compile "com.android.support:support-compat:${androidSupportVersion}"
-    compile "com.android.support:support-core-utils:${androidSupportVersion}"
-    compile "com.android.support:support-core-ui:${androidSupportVersion}"
-    compile "com.android.support:support-fragment:${androidSupportVersion}"
-    compile "com.android.support:appcompat-v7:${androidSupportVersion}"
-    compile "com.android.support:recyclerview-v7:${androidSupportVersion}"
-    compile "com.android.support:design:${androidSupportVersion}"
-    compile 'com.facebook.shimmer:shimmer:0.1.0@aar'
+    androidTestImplementation('androidx.test.espresso:espresso-core:3.1.0') {
+        exclude group: 'androidx.annotation', module: 'annotation'
+    }
 
-    testCompile 'junit:junit:4.12'
-    compile project(path: ':bricks')
+    implementation "androidx.legacy:legacy-support-v4:1.0.0"
```

Comment:
removing all `legacy` dependencies should be fine.

Reply:
It didn't break the demo app at least!

Reply:
Wouldn't break anything. But also not necessary. For app, it is not so important but for the library it might be important. 

Keep in mind that every dependency you use goes into clients of the library and keeps space.